core/src/main/java/com/graphhopper/routing/weighting/custom/SpeedCustomConfig.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.weighting.custom;
+
+import com.graphhopper.routing.profiles.BooleanEncodedValue;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.profiles.EncodedValueLookup;
+import com.graphhopper.routing.profiles.EnumEncodedValue;
+import com.graphhopper.routing.util.CustomModel;
+import com.graphhopper.util.EdgeIteratorState;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.prep.PreparedGeometryFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.graphhopper.routing.weighting.custom.PriorityCustomConfig.createEnumToDoubleArray;
+import static com.graphhopper.routing.weighting.custom.PriorityCustomConfig.getEV;
+
+final class SpeedCustomConfig {
```

Comment:
Can I rename this to `SpeedCalculator`? I think that it is what this thing does. It can be configured by the custom model, but what it does is calculating speeds (as needed by custom weighting). Its not a configuration. If anything is a configuration then the `CustomModel` (which we might want to call `CustomWeightingConfig` in correspondence to `ProfileConfig`) to not introduce more vocabulary (model)).

Reply:
> Can I rename this to SpeedCalculator?

Sure. Should we rename also to PriorityCalculator?

> CustomWeightingConfig

Yes, good idea!

Reply:
> Sure. Should we rename also to PriorityCalculator?

Yes of course :)

Reply:
> CustomWeightingConfig

Ok, I will try how this makes sense in the documentation as well (we also call this the 'model' there)